detox/src/devices/drivers/android/AttachedAndroidDriver.js

```diff
@@ -14,24 +23,35 @@ class AttachedAndroidDriver extends AndroidDriver {
   }
 
   async acquireFreeDevice(deviceQuery) {
-    const adbName = _.isPlainObject(deviceQuery) ? deviceQuery.adbName : deviceQuery;
-    const { devices, stdout } = await this.adb.devices();
-
-    if (!devices.some(d => d.adbName === adbName)) {
-      throw new DetoxRuntimeError({
-        message: `Could not find '${adbName}' on the currently ADB attached devices:`,
-        debugInfo: stdout,
-        hint: `Make sure your device is connected.\n` +
-              `You can also try restarting adb with 'adb kill-server && adb start-server'.`
-      });
-    }
+    const adbNamePattern = _.isPlainObject(deviceQuery) ? deviceQuery.adbName : deviceQuery;
+
+    const adbName = await this._allocateDevice(adbNamePattern);
 
     await this.adb.apiLevel(adbName);
     await this.adb.unlockScreen(adbName);
+    await this.emitter.emit('bootDevice', { deviceId: adbName });
```

Comment:
@noomorph does this make sense here?

Reply:
This step is necessary for the artefact managers to [register](https://github.com/wix/Detox/blob/master/detox/src/artifacts/templates/plugin/ArtifactPlugin.js#L106) the `deviceId`. And use it later to call ADB commands for that specific device when running in parallel mode https://github.com/wix/Detox/pull/2169#issuecomment-653050857

Reply:
👍 I'm double-checking seeing that we do not effectively _boot_ anything here